server/src/main/java/org/apache/iotdb/db/engine/merge/manage/MergeResource.java

```diff
@@ -65,24 +65,28 @@
   private Map<String, MeasurementSchema> measurementSchemaMap = new HashMap<>();
   private Map<MeasurementSchema, IChunkWriter> chunkWriterCache = new ConcurrentHashMap<>();
 
+  private long fileTimeBound = Long.MAX_VALUE;
+
   private boolean cacheDeviceMeta = false;
 
   public MergeResource(List<TsFileResource> seqFiles, List<TsFileResource> unseqFiles) {
-    this.seqFiles = seqFiles.stream().filter(res -> res.isClosed() && !res.isDeleted())
-        .collect(Collectors.toList());
+    this.seqFiles =
+        seqFiles.stream().filter(this::filterResource).collect(Collectors.toList());
     this.unseqFiles =
-        unseqFiles.stream().filter(res -> res.isClosed() && !res.isDeleted())
-            .collect(Collectors.toList());
+        unseqFiles.stream().filter(this::filterResource).collect(Collectors.toList());
+  }
+
+  private boolean filterResource(TsFileResource res) {
+    return res.isClosed() && !res.isDeleted() && res.stillLives(fileTimeBound);
```

Comment:
In the before lambda, you didn't judge res.stillLives(fileTimeBound), but in the `filterResource` you judge it. Will there be some bugs after changing it?

Reply:
No, because the default value of fileTimeBound is Long.MIN_VALUE, when I call `stillLives()`, it will return true immediately, so it equals to the previous version.

Reply:
By the way, if you are satisfied with my answers, please resolve the conversations.